packages/web-components/src/components/cbp-dropdown/cbp-dropdown.tsx

```diff
@@ -52,36 +56,151 @@ export class CbpDropdown {
   @Prop() sx: any = {};
 
 
+  @State() selectedItems: HTMLCbpDropdownItemElement[];
+
   /** A custom event emitted when the click event occurs for either a rendered button or anchor/link. */
   @Event() valueChange: EventEmitter;
 
-  // This handles activating the button via Space or Enter as well.
+  @Listen('dropdownItemClick')
+  handleDropdownItemClick( {detail: { host, label, value} }) {
+
+    // multi-select behavior
+    if(this.multiple) {
+      // TechDebt: this should ideally be async/promise. Update: Made selectedItems a State, so this might be fine now. Needs testing.
+      host.selected=!host.selected; // toggle the selected state of the item
+
+      setTimeout( () => {
+        this.selectedItems=Array.from(this.host.querySelectorAll('cbp-dropdown-item[selected]'));
+        this.placeholder=this.selectedItems.length != 1 ? 'Selected Items' : 'Selected Item';
+      }, 50);
+
+      this.value = []; // TODO: make an array of the values of selected items
```

Comment:
is the todo here part of the tech debt on line 69?

Reply:
updated and tested. seems to be working.